app/src/main/java/pm/gnosis/heimdall/ui/security/SecurityActivity.kt

```diff
@@ -43,49 +49,54 @@ class SecurityActivity : BaseActivity() {
     override fun onStart() {
         super.onStart()
 
-        disposables += layout_security_submit_button.clicks()
-                .flatMap {
-                    when (currentSate) {
-                        SecurityContract.State.UNINITIALIZED -> Observable.just(
-                                SecurityContract.SetupPin(
-                                        layout_security_pin_input.text.toString(),
-                                        layout_security_repeat_input.text.toString()
-                                ))
-                        SecurityContract.State.LOCKED -> Observable.just(
-                                SecurityContract.Unlock(
-                                        layout_security_pin_input.text.toString()
-                                ))
-                        else ->
-                            Observable.empty()
-                    }
-                }.compose(viewModel.transformer())
+        disposables += listen()
+                .subscribeOn(AndroidSchedulers.mainThread())
```

Comment:
Yes

Reply:
I think it is, because the flatmap moves the onsubscribe away from the main thread, but I need to make sure that the clicks onsubscribe is called on the main thread, because else it cannot set the click listener